index-scheduler/src/batch.rs

```diff
@@ -759,4 +800,55 @@ impl IndexScheduler {
             }
         }
     }
+
+    /// Delete each given task from all the databases (if it is deleteable).
+    ///
+    /// Return the number of tasks that were actually deleted
+    fn delete_matched_tasks(&self, wtxn: &mut RwTxn, matched_tasks: &[u32]) -> Result<usize> {
+        // 1. Remove from this list the tasks that we are not allowed to delete
+        let enqueued_tasks = self.get_status(wtxn, Status::Enqueued)?;
+
+        let processing_tasks = &self.processing_tasks.read().unwrap().1;
+        let to_delete_tasks = matched_tasks
+            .iter()
+            .filter(|&&task_id| {
+                !processing_tasks.contains(task_id) && !enqueued_tasks.contains(task_id)
+            })
+            .copied();
```

Comment:
Please write those two steps in a checklist:
 - don't delete tasks with an id higher or equal to the deleting task
 - use bitmaps for the tasks id ans ease operations

Reply:
Will do! :) Regarding the first point (don't delete tasks with an id higher or equal to the deleting task): since the matched tasks are determined at the time where the deletion task is created, there is very little chance that they will match a task with an id that is higher or equal to the deletion task's id. But I do agree it is better to add an extra check, especially since that guarantee might go away after a refactor.